ql/instrument.hpp

```diff
@@ -200,6 +201,22 @@ namespace QuantLib {
         return additionalResults_;
     }
 
+    //! Utility function to optimize the observability graph of an instrument
+    /*! This function unregisters the given instrument (e.g. a Swap) from the given cashflows and
+        instead registers with the observables of the cashflows. This is safe to do if
+        - the coupon pricers of the cashflows are set before the function is called and never
+          updated afterwards
+        - the cashflows are not themselves originating notifications, i.e. they only pass through
+          notifications from their observables (which is usually the case)
+        - the set of cashflows does not dynamically change (usually satisfied as well)
+        If unregisterCoupons is set to true, all given cashflows are in addition unregistered from
+        all their observables. This can be done
+        - if the coupons are not asked for results directly
+        - if deepUpdate() is called on the instrument before retrieving a result; to determine
+          whether the result might have changed, isCalculated() can be called on the instrument.
+    */
+    void passThroughNotifications(Instrument& instrument,
```

Comment:
Maybe something like `simplifyNotificationGraph` might be more descriptive?  It's probably not that clear either, but at least it says what it does instead of how it does it...